src/main/java/mate/academy/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -0,0 +1,88 @@
+package mate.academy.dao.impl;
+
+import java.util.Optional;
+import mate.academy.dao.ShoppingCartDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.User;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
+    private static final String GET_SHOPPING_CART_BY_USER_QUERY = """
+                                                FROM ShoppingCart sc 
+                                                LEFT JOIN FETCH sc.tickets t 
+                                                LEFT JOIN FETCH  sc.user 
+                                                LEFT JOIN t.movieSession ms 
+                                                LEFT JOIN t.user u 
+                                                LEFT JOIN ms.movie 
+                                                LEFT JOIN ms.cinemaHall 
+                                                WHERE sc.user = :user""";

```

Comment:
You join user twise.
```suggestion
                                                LEFT JOIN FETCH  sc.user u
                                                LEFT JOIN t.movieSession ms 
                                                LEFT JOIN ms.movie 
                                                LEFT JOIN ms.cinemaHall 
                                                WHERE u = :user""";

```